### PR TITLE
Add Triple.predicate_among.

### PR DESCRIPTION
**Pull request by ProgVal**
This method is a painless way of testing if the predicate of a predicate is in a given list/set (also works if a Resource is given instead of the list/set).

### CODE REVIEW

File: ppp_datamodel/nodes/triple.py [hunk 1]
```diff
@@ -16,3 +17,22 @@ def _check_attributes(self, attributes):
         if not all(isinstance(x, AbstractNode) for x in attributes.values()):
             raise TypeError('One of Triple\'s constructor argument '
                             'is not an AbstractNode instance.')
+
+    def predicate_among(self, L):
+        # Be quite permissive about what we accept as L and convert it to
+        # a (frozen)set.
+        if isinstance(L, AbstractNode):
+            L = frozenset({L})
+        elif not isinstance(L, (set, frozenset)):
```
Comment (Tpt) [inline ppp_datamodel/nodes/triple.py]: Is this line really needed?

Comment (ProgVal) [inline ppp_datamodel/nodes/triple.py]: No, but if I remove it I have to add an `isinstance(L, (set, frozenset))` clause to the two next conditions.
(or make a useless copy of the set)

**Comment by Tpt** [comment]
The contract of this method is not really clear. I think that the method name should be something like has_a_predicate_in because "predicate_among" may let people think that all predicates should be in the provided list/set.

"the predicate of a predicate ": I believe you mean "one of the predicates" ?

File: ppp_datamodel/nodes/triple.py (hunk 1 above)
Comment (Tpt) [inline ppp_datamodel/nodes/triple.py]: Make sense

**Comment by ProgVal** [comment]
> The contract of this method is not really clear. I think that the method name should be something like has_a_predicate_in because "predicate_among" may let people think that all predicates should be in the provided list/set.

Well, it's called “predicate_among”, not “predicates_among”…

> "the predicate of a predicate ": I believe you mean "one of the predicates" ?

Yes

**Comment by Tpt** [comment]
Well, but I'm not sure that the average user of this lib will understand. I had to read carefully the code to understand what this function was doing. So I won't merge this PR if the function name is not changed to something better.

**Comment by ProgVal** [comment]
Rewritten in a much more simple way (there was a lot of useless branches handling).

**Comment by Tpt** [comment]
Was the deletion of the predicate_among function wanted? If yes, the current change looks good to me

**Comment by ProgVal** [comment]
yes

**Pull request merged**